python/hail/dataset.py

```diff
@@ -403,7 +405,8 @@ def annotate_global(self, path, annotation, annotation_type):
         annotation_type._typecheck(annotation)
 
         annotated = self._jvds.annotateGlobal(annotation_type._convert_to_j(annotation), annotation_type._jtype, path)
-        assert annotated.globalSignature().typeCheck(annotated.globalAnnotation()), 'error in java type checking'
+        if sys.version_info < (3,):
+            assert annotated.globalSignature().typeCheck(annotated.globalAnnotation()), 'error in java type checking'
```

Comment:
The has something to do with the mapping between long which now maps to int, just like int does. I suspect we might have to remove the concept of LongType from the python side. I'll rerun it sortly. to see what the error was.